avalon/tools/models.py

```diff
@@ -336,6 +341,19 @@ def _add_hierarchy(self, assets, parent=None):
             None
 
         """
+        if silos:
+            # WARNING: Silo item "_id" is set to silo value
+            # mainly because GUI issue with perserve selection and expanded row
```

Comment:
1. `perserve` -> `preserve`

2. What exactly is the issue described here? What in particular caused that issue? Is that the case when the "silo" is *automagically* added based on an asset's "data.silo" value and as such it would have the same `_id` as the asset? Either way, good to see this clearly "warned" in a comment.

Reply:
1.) I will never be a copywriter :(

2.) It is because silo can be used as parent in `assets` argument for assets with `visualParent` set to None.
And `preserve_states` may use `ObjectIdRole` which means it is better to have set `_id` to any other value then `None` (because each silo item would have _id value set to None). Makes that sence?